lexers/rest.lua

```diff
@@ -135,38 +95,32 @@ local underline = lpeg.Cmt(starts_line(adornment), function(_, index, adm, c)
 	return pos and index - #adm + pos - 1 or nil
 end)
 -- Token needs to be a predefined one in order for folder to work.
-lex:add_rule('title', token(lexer.HEADING, overline + underline))
+lex:add_rule('title', lex:tag(lexer.HEADING, overline + underline))
 
 -- Line block.
-lex:add_rule('line_block_char', token(lexer.OPERATOR, starts_line(any_indent * '|')))
-
--- Whitespace.
-lex:add_rule('whitespace', token(lexer.WHITESPACE, S(' \t')^1 + lexer.newline^1))
```

Comment:
The migration guide allows for lexers to define their own whitespace rules if whitespace is significant. That is the case here. However, reST is a special case in that its whitespace rule is below other rules (normally it's at the top, like the Markdown lexer). I honestly don't know how this is going to work in a migrated lexer.

Regardless, removing this rule will break the lexer. We need it, I'm just not sure how.

Reply:
Readded